cmake/modules/board_extensions.cmake

```diff
@@ -0,0 +1,32 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) 2023, Teslabs Engineering S.L.
+
+# This module will look for board extensions.
+#
+# Outcome:
+# The following variables will be defined when this module completes:
+# - board_extension_conf_files: List of Kconfig fragments
+# - board_extension_dts_files : List of devicetree files
+#
+# Optional variables:
+# - BOARD_ROOT: CMake list of board roots containing board implementations or
+#   extensions.
+#
+# Variables set by this module and not mentioned above are for internal
+# use only, and may be removed, renamed, or re-purposed without prior notice.
+
+include_guard(GLOBAL)
+
+include(extensions)
+
+foreach(root ${BOARD_ROOT})
+  set(board_extension_dir ${root}/boards/extensions/${BOARD})
```

Comment:
This creates differing behaviour to what zephyr has currently - assume using the board `nrf5340dk_nrf5340_cpuapp_ns` - this board is located in `<zephyr>/boards/arm/nrf5340dk_nrf5340` but for an extension, puts files in `<zephyr>/boards/extensions/nrf5340dk_nrf5340_cpuapp_ns`

Reply:
good point, I've updated to obtain the directory name from `BOARD_DIR` instead.